src/otx/v2/adapters/torch/dataset.py

```diff
@@ -0,0 +1,198 @@
+"""OTX adapters.torch.mmengine.mmpretrain.Dataset API."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Iterable
+
+import yaml
+from torch.utils.data import DataLoader as TorchDataLoader
+from torch.utils.data import Dataset as TorchDataset
+from torch.utils.data import Sampler
+
+from otx.v2.api.core.dataset import BaseDataset
+from otx.v2.api.utils import set_tuple_constructor
+from otx.v2.api.utils.decorators import add_subset_dataloader
+
+SUBSET_LIST = ["train", "val", "test"]
+
+
+@add_subset_dataloader(SUBSET_LIST)
+class TorchBaseDataset(BaseDataset):
+    """A class representing a dataset for pretraining a model."""
+
+    def _initialize(self) -> None:
+        self.set_datumaro_adapters()  # Set self.dataset_entity & self.label_schema
+        self.initialize = True
+
+    def build_dataset(
+        self,
+        subset: str,
+        pipeline: list | dict | None = None,
+        config: str | dict | None = None,
+    ) -> TorchDataset | None:
+        """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (Optional[Union[list, dict]]): The pipeline to use for the dataset.
+                Defaults to None.
+            config (Optional[Union[str, dict]]): The configuration to use for the dataset.
+                Defaults to None.
+
+        Returns:
+            Optional[TorchDataset]: The built TorchDataset object, or None if the dataset is empty.
+        """
+
```

Comment:
AFAIU, this would be used as an interface? could we add raise NotImplementedError?